docs/development-guide.md

```diff
@@ -0,0 +1,192 @@
+# Development Guide
+
+## New to Go
+
+Ceph-csi is written in Go and if you are new to the language,
+it is **highly** encouraged to:
+
+* Take the [A Tour of Go](http://tour.golang.org/welcome/1) course.
+* [Set up](https://golang.org/doc/code.html) Go development environment on your machine.
+* Read [Effective Go](https://golang.org/doc/effective_go.html) for best practices.
+
+## Development Workflow
+
+### Workspace and repository setup
+
+* [Download](https://golang.org/dl/) Go and
+   [install](https://golang.org/doc/install) it on your system.
+* Run
+
+    ```console
+    go get -d github.com/ceph/ceph-csi-operator
+    ```
+
+   This will just download the source and not build it. The downloaded source
+   will be at `$GOPATH/src/github.com/ceph/ceph-csi-operator`
+* Fork the [ceph-csi-operator repo](https://github.com/ceph/ceph-csi-operator) on Github.
+* Add your fork as a git remote:
+
+    ```console
+    git remote add fork https://github.com/<your-github-username>/ceph-csi-operator
+    ```
```

Comment:
As a development guide, why are we advocating users to run `go get` into their global GOPATH?
As it stands today, most new projects are trying to stay away from non standard lib dependencies in GOPATH

Reply:
update code to remove go get and clone the repo